code/obj/submachine/weapon_vendor.dm

```diff
@@ -213,6 +213,16 @@
 		STOP_TRACKING_CAT(TR_CAT_NUKE_OP_STYLE)
 		..()
 
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		. = ..()
+		if (.)
```

Comment:
I would check for the param rather than the return value here, just in case someone adds another action to the parent proc.